### PR TITLE
pwx-34665 | test: add tests for failover workflow in sync dr

### PR DESCRIPTION
**Pull request by strivedi-px**
**What type of PR is this?**
>integration-test

**What this PR does / why we need it**:
The PR adds integration test for the failover workflow in the Sync/Metro DR setup.

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:
No

**Does this change need to be cherry-picked to a release branch?**:
Yes, to `release-24.2.0` branch.

**Test runs**
- Pipeline run: https://jenkins.pwx.dev.purestorage.com/job/Users/job/strivedi/job/strivedi-cluster-domain/3/console
- Local run
Screenshot from 2024-05-21 21-02-20

### CODE REVIEW

File: test/integration_test/common_test.go [hunk 1]
```diff
@@ -838,7 +838,7 @@ func scheduleClusterPair(ctx *scheduler.Context, skipStorage, resetConfig bool,
 }
 
 // Create a cluster pair from source to destination and another cluster pair from destination to source
-func scheduleBidirectionalClusterPair(cpName, cpNamespace, projectMappings string, objectStoreType storkv1.BackupLocationType, secretName string) error {
+func scheduleBidirectionalClusterPair(cpName, cpNamespace, projectMappings string, objectStoreType storkv1.BackupLocationType, secretName string, isMetroDR bool) error {
```
Comment (diptiranjanpx) [inline test/integration_test/common_test.go:841]: Let's name the variable as `isSyncDR`

File: test/integration_test/storkctl_dr_action_test.go [hunk 2]
```diff
@@ -286,6 +286,23 @@ func getDestinationKubeConfigFile() (string, error) {
 	return destKubeconfigPath, err
 }
 
+func getSourceKubeConfigFile() (string, error) {
```
Comment (diptiranjanpx) [inline test/integration_test/storkctl_dr_action_test.go:289]: Let's move this to utils.

Comment (strivedi-px) [reply test/integration_test/storkctl_dr_action_test.go]: Done

File: test/integration_test/common_test.go (hunk 1 above)
Comment (strivedi-px) [reply test/integration_test/common_test.go]: Done

File: test/integration_test/storkctl_dr_action_test.go (hunk 2 above)
Comment (diptiranjanpx) [reply test/integration_test/storkctl_dr_action_test.go]: My bad, I meant common_test.go not pkg/utils. Let's keep the common functions used for test here.

Comment (strivedi-px) [reply test/integration_test/storkctl_dr_action_test.go]: Done. Please take another look.

**Review by diptiranjanpx — APPROVED** [review]

**Review by diptiranjanpx — APPROVED** [review]
LGTM

**Pull request merged**